<?xml version="1.0"?>
<launch>
    <arg name="visual3d_path" default="/home/visualDet3D_open"/>
    <arg name="cfg_file" default="/home/visualDet3D_open/config/kitti_yolo3d_copied.py"/>
    <arg name="weight_path" default="/home/visualDet3D_open/workdirs/Mono3D/checkpoint/GroundAwareYolo3D_latest.pth"/>
    <arg name="camera_topic" default="/kitti/left_camera/image"/>
    <arg name="camera_param" default="/kitti/left_camera/camera_info"/>
    <arg name="bbox_topic" default="/bboxes"/>
    <arg name="CROP_TOP" default="100"/>
    <arg name="INFERENCE_SCALE" default="1.0"/>
    <arg name="INFERENCE_W" default="1280"/>
    <arg name="INFERENCE_H" default="288"/>

    <node pkg="yolo3d" type="yolo3d_node.py" name="yolo3d_node" output="screen">
        <param name="VISUAL3D_PATH" value="$(arg visual3d_path)"/>
        <param name="CFG_FILE" value="$(arg cfg_file)"/>
        <param name="WEIGHT_PATH" value="$(arg weight_path)"/>
        <param name="CROP_TOP" value="$(arg CROP_TOP)"/>
        <param name="INFERENCE_SCALE" value="$(arg INFERENCE_SCALE)"/>
        <param name="INFERENCE_W" value="$(arg INFERENCE_W)"/>
        <param name="INFERENCE_H" value="$(arg INFERENCE_H)"/>
        <remap from="/image_raw" to="$(arg camera_topic)"/>
        <remap from="/camera_info" to="$(arg camera_param)"/>
        <remap from="/bboxes" to="$(arg bbox_topic)"/>
    </node>

</launch>